HW0/P4/P4.py

```diff
@@ -24,53 +24,68 @@
     # Create 3D axes for plotting
     ax = Axes3D(plt.figure())
 
-    #####################
+
     # Part 1:
-    #
-    # Load true trajectory and plot it
-    # Normally, this data wouldn't be available in the real world
-    #####################
+    s_true=np.loadtxt('P4_trajectory.txt',delimiter=',')
+    true_x=s_true[:,0]
+    true_y=s_true[:,1]
+    true_z=s_true[:,2]
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    ax.plot(true_x, true_y, true_z,'--b', label='True trajectory')
 
-    #####################
+    
     # Part 2:
-    #
-    # Read the observation array and plot it (Part 2)
-    #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    s_measure=np.loadtxt('P4_measurements.txt',delimiter=',')
+    measure_x=s_measure[:,0]/rx
+    measure_y=s_measure[:,1]/ry
+    measure_z=s_measure[:,2]/rz
+    
+    ax.plot(measure_x, measure_y, measure_z,'.g', label='Observed trajectory')
 
-    #####################
     # Part 3:
-    # Use the initial conditions and propagation matrix for prediction
-    #####################
-
-    # A = ?
-    # a = ?
-    # s = ?
+    A=np.array([[1,0,0,dt,0,0],[0,1,0,0,dt,0],[0,0,1,0,0,dt],[0,0,0,1-c*dt,0,0],[0,0,0,0,1-c*dt,0],[0,0,0,0,0,1-c*dt]])
```

Comment:
indenting out line-by-line will make it easier for debugging and the reader.
[1,0,0,dt,0,0],
[0,1,0,0,dt,0],
[0,0,1,0,0,dt],
[0,0,0,1-c*dt,0,0],
[0,0,0,0,1-c*dt,0],
[0,0,0,0,0,1-c*dt]])